### DISCUSSION TITLE
Fine details rendering too thin/light to print

### DISCUSSION THREAD

**Discussion by SwervingLemon** [sumatrapdfreader/sumatrapdf#2781] [Q&A]
Schematics printed in Sumatra are often illegible, prints look more accurate to the screen display than through other viewers, but at the cost of visibility of fine lines.
Have tried adjusting "fit" "stretch" "fill" etc. in the print dialogues but to no avail.
Have tried adjusting settings in the printer but this problem seems unique to Sumatra, as the same pages printed from either a browser or Adobe's own PDF viewer still manage to print the circuit trace lines.

I think there's a feature in the print renderer that's not being addressed, like a minimum line thickness or path min stroke or something. Just my WAG.

**Comment by GitHubRulesOK** [comment]
Usually is an issue with the authoring not setting ink wide enough so thinnest line for ISO A4 (similar to letter)  1 printing point = 0.35mm thus 0.18 mm ISO pen = 1/2 point so should be optimal and visible in printing at around 150 pixels grey per inch or at 75dpi should become blacker but Microsoft printing may thin it down to 1/4 of optimal (600dpi) either too pale or invisible.

Acrobat printing will modify the inked line thickness for viewing or printing. see https://github.com/sumatrapdfreader/sumatrapdf/issues/1732#

If you post an example I can see how much thicker is needed but it is the writer that needs to be adjusted

**Comment by SwervingLemon** [comment]
I was hoping to have found something similar and I did search for similar questions but I guess my phrasing was distinct enough (or case different enough) that 1732 didn't show. Maybe it's because I'm specifically focused on printing.
However:
Generally, I don't care if Sumatra is "closer to spec". I only care whether or not it produces usable prints, as every other reader does.
In every other respect, Sumatra is my hands-down favorite. Unfortunately, it's "proper" handling of line thicknesses renders it unusable for my purposes, and I certainly can't get the manufacturers to revise literally hundreds of manuals just because my reader of choice renders the lines "correctly".
Sumatra needs an option to make it's output to the print engine the same or similar to what is accepted and normal practice, even if it's "less accurate".

Here's a photo, with Sumatra on the left and Adobe on the right, both set to "actual size" yet the Adobe display is larger and has _thinner_ lines which actually print, while Sumatra's display of the same "actual size" is smaller (which is correct, Adobe has the screen dpi messed up somehow, apparently), has _broader_ lines showing and the lines don't print:

![SumatraVsAdobe](https://user-images.githubusercontent.com/41270988/177205100-2d22a153-28b8-4e28-b0bc-b22e54051e71.png)

Those are side-by-side windows on my primary display, right now, one screen grab and nothing's been resized.
Both of those are set to "actual size" for US Letter. Why the disparity in size?

At 200% zoom, the reason for the print difference is more obvious. The display engine on Sumatra does less antialiasing or whatever it was trying to do to display the thin lines and we get _this_ result:

![SumatraVsAdobe2](https://user-images.githubusercontent.com/41270988/177206507-e7b6bb4b-814e-44c0-baf8-a1a1e2ecaa91.png)

**Reply by GitHubRulesOK** [reply]
All setting equal at 100% SumatraPDF is possibly thinner / cleaner
![image](https://user-images.githubusercontent.com/12754913/177218301-e3064c25-280f-4cb5-9a92-22a0465687a9.png)

Printing in SumatraPDF is 100 % down to windows so without repeating previous paper results by way of explanation what happens is down to drivers so here the current driver I have active will output this thermal image, which is perfectly normal for that Microsoft input/Output since grey is converted to castellated edges without any anti-alias. Adobe are Print Driver Producers so they like other Print Driver writers can adapt the printing output as they see fit.

![image](https://user-images.githubusercontent.com/12754913/177218872-7c3c7738-f921-488d-b46b-bfc6919013b2.png)

MicroSoft own apps drivers and Acrobat can aggregate the output, Hence for thermals its best using different drivers.

![image](https://user-images.githubusercontent.com/12754913/177335551-12db34da-bed6-4aa8-a186-ac9ca363ed3a.png)

**Reply by SwervingLemon** [reply]
I found and changed the custom dpi on both programs to match the native resolution of my monitor and now the displays are actually 1:1, which is cool...

![SumatraVsAdobe3](https://user-images.githubusercontent.com/41270988/177365678-2af4ae67-72f3-4e94-a074-fc7cc081fc95.png)

And I've found that if I change my print default in Sumatra to "fit" instead of "shrink", many of the previously unreadable lines are now at least faintly visible. It's still far from easily legible prints but at least they're not completely unusable.
